grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1536,6 +1559,12 @@ class StockMovementService {
         if (shipment && shipment.currentStatus > ShipmentStatusCode.PENDING) {
             shipmentService.rollbackLastEvent(shipment)
             requisitionService.rollbackRequisition(requisition)
+            requisition.requisitionItems.each { requisitionItem ->
+                if (requisitionItem.orderItem) {
+                    OrderItem orderItem = OrderItem.get(requisitionItem.orderItem.id)
+                    orderService.updateOrderItem(orderItem, -requisitionItem.quantity)
```

Comment:
What if quantityFulfilled will be lower than requisitionItem.quantity, then we will have value lower than 0

Reply:
QuantityFulfilled of order item is a sum of all quantities of requisition items having reference to this order item. During rollback we just reduce quantityFullfiled to 0 by iterating over requisition items.